core/domain/classifier_services.py

```diff
@@ -127,6 +127,52 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+def create_classifier_training_jobs(exploration, state_names):
+    """Creates ClassifierTrainingJobModel instances for all the state names
+    passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration Domain object.
+        state_names: list. List of state names.
+    """
+    exp_id = exploration.id
+    exp_version = exploration.version
+    states = exploration.states
+    for state_name in state_names:
+        state = states[state_name]
+        training_data = state.get_training_data()
+        interaction_id = state.interaction.id
+        algorithm_id = feconf.INTERACTION_CLASSIFIER_MAPPING[
+            interaction_id]['algorithm_id']
+        create_classifier_training_job(algorithm_id, interaction_id,
+                                       exp_id, exp_version, state_name,
+                                       training_data,
+                                       feconf.TRAINING_JOB_STATUS_NEW)
+
+
+def update_classifier_exploration_mappings(exploration, state_names):
+    """Creates new ClassifierExplorationMappingModel instances for all the
+    state names passed into the function.
+
+    Args:
+        exploration: Exploration. The Exploration Domain object.
+        state_names: list. List of state names.
+    """
+    exp_id = exploration.id
```

Comment:
@seanlip Yep. We can do the checks for the previous classifier in that way.

Re your query: If that happens, we delete the job (since its not picked up by the VM yet, and create a new job for the new version.)